homeassistant/components/homewizard_energy/coordinator.py

```diff
@@ -0,0 +1,93 @@
+"""Update coordinator for HomeWizard Energy."""
+
+from datetime import timedelta
+import logging
+
+import aiohwenergy
+import async_timeout
+
+from homeassistant.const import CONF_API_VERSION, CONF_ID, CONF_NAME, CONF_STATE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    ATTR_BRIGHTNESS,
+    ATTR_POWER_ON,
+    ATTR_SWITCHLOCK,
+    CONF_DATA,
+    CONF_MODEL,
+    CONF_SW_VERSION,
+    MODEL_P1,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HWEnergyDeviceUpdateCoordinator(DataUpdateCoordinator):
+    """Gather data for the energy device."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        api: aiohwenergy.HomeWizardEnergy,
+    ) -> None:
+        """Initialize Update Coordinator."""
+
+        self.api = api
+
+        update_interval = self.get_update_interval()
+        super().__init__(hass, _LOGGER, name="", update_interval=update_interval)
+
+    def get_update_interval(self) -> timedelta:
+        """Return best interval for product type."""
+        try:
+            product_type = self.api.device.product_type
+        except AttributeError:
+            product_type = "Unknown"
+
+        if product_type == MODEL_P1:
+            try:
+                smr_version = self.api.data.smr_version
+                if smr_version == 50:
+                    return timedelta(seconds=1)
+
+            except AttributeError:
+                pass
+
+        return timedelta(seconds=5)
+
+    async def _async_update_data(self) -> dict:
+        """Fetch all device and sensor data from api."""
+        try:
+            async with async_timeout.timeout(10):
+                # Update all properties
+                status = await self.api.update()
+
+                if not status:
+                    raise Exception("Failed to fetch data")
```

Comment:
Raise `UpdateFailed` directly and drop the try…except

Reply:
Changed with [d7a09d6c](https://github.com/DCSBL/core/commit/d7a09d6c5feb8f31589aa2e2294418fd3dbcc22d). But I think we still need to catch the [timeout](https://github.com/DCSBL/core/blob/d7a09d6c5feb8f31589aa2e2294418fd3dbcc22d/homeassistant/components/homewizard_energy/coordinator.py#L47)?

Reply:
[Nope](https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/update_coordinator.py#L187-L189)

Reply:
O nice! Removed the try..catch 👍 https://github.com/home-assistant/core/pull/55812/commits/1688b2826f8b2a1d2feef52a9d49f4aa23026986